### DISCUSSION TITLE
Blocking all mastodon instances

### DISCUSSION THREAD

**Discussion by xcrunner8820** [mastodon/mastodon#29377] [Q&A]
Hi everyone, 

I know this sounds very weird on an official mastodon page, but I'm trying to block all social media sites on my computer (because of mental health and impulse control), and mastodon is proving to be incredibly difficult to block because there are so many different instances that use different names. Does anyone know a way to block mastodon in general? Thank you in advance and I know this is a weird ask.

**Comment by MadokVaur** [comment]
I probably should avoid this, but I don't even understand the question

How do you even *see* Mastodon or any "social media sites" on your computer in the first place?

Via a web browser? Delete bookmarks. Betting you created them in the first place

Some sort of "apps"? Delete them. Betting you installed them in the first place

What am I missing?

**Reply by xcrunner8820** [reply]
Yeah I definitely get your answer. I'd say what you're missing is the impulse control part. I want it to be out of my hands to be able to get at social media. Deleting the apps or the bookmarks don't stop me from being able to get at the site

**Reply by MadokVaur** [reply]
"I'd say what you're missing is the impulse control part"

No

I saw that

I ignored it, for obvious reasons

You want a parent, not Mastodon technical support

**Reply by xcrunner8820** [reply]
I'm literally working with them to try and configure our internet filter to block all the instances of mastodon. But blocking "mastodon.social" does not block all the instances. So the question is, What do I enter into our internet filter in order to filter out all instances of mastodon? I'm asking here because you guys are the SMEs on your own site.

**Reply by michaelmannucci** [reply]
> I'm literally working with them to try and configure our internet filter to block all the instances of mastodon. But blocking "mastodon.social" does not block all the instances. So the question is, What do I enter into our internet filter in order to filter out all instances of mastodon? I'm asking here because you guys are the SMEs on your own site.

Wow, what a jerk.

Anyways, I know it's late, but for anyone else who stumbles upon this unfortunate interaction: if you use Chrome, I believe there are multiple extensions that can block anything that contains certain keywords. If you want to get real fancy with it, some registry edits can make it so those extensions can't be removed.

If you have an iPhone, I believe NextDNS can also block access to certain keywords.

This is in theory, I have tried neither.

**Reply by R2Pitou** [reply]
### I’m late to the game, but here’s my 2 cents as a Digital Transformation Supervisor in a K‑12 school:

We don’t lose money if students hop on Instagram… we lose learners to endless, low‑effort dopamine pops delivered from outside classroom walls (The barricades that used to keep distractions at bay.) Reading this thread, I’m concerned by the attitude that Mastodon “can’t” or “shouldn’t” be blocked... even consensually. Some here seem more focused on federating their farts (or opinions; I keep forgetting which one it is) rather than addressing the very real issue of social media as addiction and distraction.

Are there unique CSS classes that are always in a certain pattern that could be used to determine if a site is indeed a Mastodon node? (Wildly pulling ideas out of thin air.)

[Here’s some great wind from Tim Minchin](https://www.youtube.com/watch?v=G1juPBoxBdc) Not strictly on topic, but please let a teacher use this teachable moment for a useful tangent 🙇

**Comment by technicat** [comment]
you can determine a site is mastodon or other fediverse platform by checking if it has a nodeinfo (that's what many client apps do)

https://github.com/jhass/nodeinfo

**Comment by nclm** [comment]
You could try to get a full list from [instances.social](https://instances.social) maybe.

**Comment by Stacey-R** [comment]
For quick productivity / mental health blocking that's still in your control, I use https://www.proginosko.com/leechblock/

There's probably an add on for whatever browser you are using.

**Comment by nenben1lol** [comment]
Hi @xcrunner8820 — I’m trying to do something similar: block Mastodon in a way I can’t easily disable myself.

My plan is DNS-level blocking with an admin-installed macOS profile. I tried NextDNS first, then AdGuard DNS because it supports custom blocklists.

I found a list of ~17,000 Mastodon instances, but AdGuard counts every domain as a separate rule, so I immediately hit the plan’s rule limit.

**Reply by nenben1lol** [reply]
I was able to add the ~17,000-domain Mastodon instance list to AdGuard as a custom blocklist, so technically the approach works. The issue is that AdGuard counts each domain as a separate rule, which exceeds the normal plan limits.

The Enterprise plan would solve that, but I’m hoping there’s a better or more efficient solution. Maybe someone here has an idea?

**Reply by nenben1lol** [reply]
I found a workable solution and ended up going back to **NextDNS**.

NextDNS doesn’t natively support arbitrary custom blocklist URLs, but its denylist can handle a very large number of domains. To bridge that gap, I used the open-source [[NextDNS Blocklist Automator](https://github.com/DragonOfDistantLands/nextdns-blocklist-automator)](https://github.com/DragonOfDistantLands/nextdns-blocklist-automator) and ran it locally on my Mac.

I imported these two Mastodon instance lists:

* [[mastodon_1-9999.txt](https://raw.githubusercontent.com/m4nbat/mastodon_servers/main/mastodon_1-9999.txt)](https://raw.githubusercontent.com/m4nbat/mastodon_servers/main/mastodon_1-9999.txt)
* [[mastodon_10000-17175.txt](https://raw.githubusercontent.com/m4nbat/mastodon_servers/main/mastodon_10000-17175.txt)](https://raw.githubusercontent.com/m4nbat/mastodon_servers/main/mastodon_10000-17175.txt)

Together they contain roughly 17,000 Mastodon instance domains.

My setup now is:

* NextDNS via an admin-installed macOS DNS profile
* the ~17,000 Mastodon domains synced into the NextDNS denylist
* the Blocklist Automator run locally, so I don’t have to enter my NextDNS API key into a hosted third-party service
* the DNS profile is protected by the Mac admin password, which is held by someone else

This avoids the AdGuard DNS limitation I ran into, where every imported domain counted as an individual custom rule and exceeded the normal plan limits.

For future updates, I can just run the Blocklist Automator again and resync the two source lists.
